tools/clang/test/HLSL/InputPatch-const.hlsl

```diff
@@ -21,9 +16,9 @@ cbuffer b : register(b0) {
 [instance(24)]
 void main(InputPatch<GSOut, 2>points, inout PointStream<GSOut> stream) {
 
-  points[0].norm[0] = 1;
-  points[0].norm[1] = 2;
+  points[0].norm[0] = 1; // expected-error {{read-only variable is not assignable}}
+  points[0].norm[1] = 2; // expected-error {{read-only variable is not assignable}}
```

Comment:
I don't think points should be considered const.  Or is our subscript operator returning the wrong type here?  I believe this is incompatible with fxc with this change.

Reply:
The docs for InputPatch’s operator[] say “ Gets a read-only resource variable.”, and our implementation returns const.

If that’s not correct we can fix our InputPatch implementation, but that should be orthogonal to this change.

See: https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/sm5-object-inputpatch

Reply:
I just checked and it turns out FXC doesn't like this code either.  Strange as I thought I'd seen this pattern somewhere, and thought this came from some sample (since it's under a path with that implication).  So it's all good!